.ai/auto-review/github_reviewer.py

```diff
@@ -92,11 +93,31 @@ def main():
             for response in responses:
                 if response.line:
                     result = post_line_comment(github=github, file=file, text=response.text, line=response.line)
+                if not result:
+                    diff_hunks = extract_diff_hunks(file_diffs)
+                    result = post_line_comment_with_diff_hunks(github=github, file=file, text=response.text, line=response.line, diff_hunk=diff_hunks)
                 if not result:
                     result = post_general_comment(github=github, file=file, text=response.text)
                 if not result:
                     raise RepositoryError("Failed to post any comments.")
 
+def extract_diff_hunks(diff_text, context_lines=3):
+    hunks = []
+    # Match each hunk starting with @@ and capture the line and surrounding context
+    hunk_pattern = re.compile(r'(@@ .+ @@)([\s\S]*?)(?=(^@@|^diff|\Z))', re.MULTILINE)
+
+    for match in hunk_pattern.finditer(diff_text):
```

Comment:
109 : Potential IndexError: `hunk_body[:context_lines + 1]` may raise an `IndexError` if `context_lines` is larger than the length of `hunk_body`: Add a check for the length of `hunk_body` or use slicing with appropriate error handling, like `hunk_body[:min(context_lines + 1, len(hunk_body))]`.